File: netkaja-app/src/app/classes/food.ts
import { Allergen } from '../classes/allergen';

export class Food {
    id: number;
    type: string;
    price: number;
    name: string;
    allergens: Array<Allergen>;

    constructor(
       
    ) { }

    public getId(): number{
        return this.id;
    }
}

File: src/main/java/com/bead/netkaja/model/Food.java
package com.bead.netkaja.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Food extends Base{
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private FoodType foodType;

    public enum FoodType {
        APPERTIZER,MAINCOURSE,DESSERT;
    }

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @ManyToMany
    private List<Allergen> allergens;
}
